portfolio/src/main/java/com/google/sps/data/BaseTaskScheduler.java

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.Collection;
+
+// TODO: Add test coverage for scheduling algorithms
+public abstract class BaseTaskScheduler {
```

Comment:
I think this class can be package-private (i.e., take out the public keyword).

Reply:
I don't think I can do that because I use this class in the schedule servlet which is in a different package.

Reply:
I think you don't want to reference BaseTaskScheduler outside this package. What you can do is create an interface that is public, with the two methods. Then you can have the task scheduler classes implement that interface, and reference the interface elsewhere rather than this abstract class.

Reply:
In hindsight, some of the stuff I'm doing in this servlet makes it impossible to have schedulers be package-private. As a result I created a class called ServletUtils  to 1. clean up the servlet a bit 2. allow the schedulers to be package-private. All I did was copy over the existing helper methods and put them into that class.